src/libs/shared/src/common/application/services/artifact_service.rs

```diff
@@ -299,4 +301,44 @@ impl ArtifactService {
 
         Ok((artifact.id.to_string(), stacker))
     }
+
+    pub async fn find_artifact_by_artifact_id(&self, artifact_id: UuidOrString) -> Result<Option<Artifact>, ArtifactServiceError> {
+        let artifact_id = match artifact_id {
+            UuidOrString::Uuid(id) => id,
+            UuidOrString::String(id) => match Uuid::parse_str(&id) {
+                Ok(uuid) => uuid,
+                Err(_) => return Err(ArtifactServiceError::NotFound(format!("Invalid UUID string: {}", id)))
+            }
+        };
+
+        // Closure for fetching the artifact
+        let find_artifact = || self.artifact_repo.find_by_id(artifact_id);
+
+        // Find the artifact
+        let maybe_artifact = retry_async(find_artifact, &Self::REPO_RETRY_POLICY).await
+            .map_err(|err| ArtifactServiceError::RepoError(err))?;
+
+        let artifact = match maybe_artifact {
+            Some(a) => a,
+            None => {
+                GlobalLogger::error(format!("Cannot find any record of the Artifact associated with ID '{}'.", artifact_id).as_str());
+                return Err(ArtifactServiceError::NotFound("Cannot find any record of the artifact associated with ID".into()))
+            }
+        };
+
+        Ok(Some(artifact))
+    }
+
+    pub async fn download_artifact(&self, input: DownloadArtifactInput) -> Result<PathBuf, ArtifactServiceError> {
```

Comment:
I do not think this is named correctly. Although I understand you intention behind this method is to get the path to the artifact so that you can download it, there is no indication that anything related to downloading is happening in this function. You should rename it or get rid of it all together. Perhaps rename it to something thhat explains the intent better like "get_downloadable_artifact". Return an error if path is missing or if it is not missing, return the whole artifact. Or better yet, and output that contains all of the details you need to download the artifact